ravenframework/Optimizers/RavenSampled.py

```diff
@@ -334,36 +356,63 @@ def finalizeSampler(self, failedRuns):
+
     # further check active unfinished trajectories
     # FIXME why should there be any active, unfinished trajectories when we're cleaning up sampler?
     traj = 0 # FIXME why only 0?? what if it's other trajectories that are active and unfinished?
-    # sanity check: if there's no history (we never got any answers) then report than rather than crash
+    # sanity check: if there's no history (we never got any answers) then report rather than crash
     if len(self._optPointHistory[traj]) == 0:
       self.raiseAnError(RuntimeError, f'There is no optimization history for traj {traj}! ' +
                         'Perhaps the Model failed?')
-    opt = self._optPointHistory[traj][-1][0]
-    val = opt[self._objectiveVar]
-    self.raiseADebug(statusTemplate.format(status='active', traj=traj, val=s * val))
-    if bestValue is None or val < bestValue:
-      bestValue = val
-      bestTraj = traj
-    bestOpt = self.denormalizeData(self._optPointHistory[bestTraj][-1][0])
-    bestPoint = dict((var, bestOpt[var]) for var in self.toBeSampled)
-    self.raiseADebug('')
-    self.raiseAMessage(' - Final Optimal Point:')
-    finalTemplate = '    {name:^20s}  {value: 1.3e}'
-    finalTemplateInt = '    {name:^20s}  {value: 3d}'
-    self.raiseAMessage(finalTemplate.format(name=self._objectiveVar, value=s * bestValue))
-    self.raiseAMessage(finalTemplateInt.format(name='trajID', value=bestTraj))
-    for var, val in bestPoint.items():
-      self.raiseAMessage(finalTemplate.format(name=var, value=val))
-    self.raiseAMessage('*' * 80)
-    # write final best solution to soln export
-    self._updateSolutionExport(bestTraj, self.normalizeData(bestOpt), 'final', 'None')
+
+    if not self._isMultiObjective:
+      opt = self._optPointHistory[traj][-1][0]
+      val = opt[self._objectiveVar[0]]
+      self.raiseADebug(statusTemplate.format(status='active', traj=traj, val=s * val))
+      if bestValue is None or val < bestValue:
+        bestValue = val
+        bestTraj = traj
+      bestOpt = self.denormalizeData(self._optPointHistory[bestTraj][-1][0])
+      bestPoint = dict((var, bestOpt[var]) for var in self.toBeSampled)
+      self.raiseADebug('')
+      self.raiseAMessage(' - Final Optimal Point:')
+      finalTemplate = '    {name:^20s}  {value: 1.3e}'
+      finalTemplateInt = '    {name:^20s}  {value: 3d}'
+      self.raiseAMessage(finalTemplate.format(name=self._objectiveVar[0], value=s * bestValue))
+      self.raiseAMessage(finalTemplateInt.format(name='trajID', value=bestTraj))
+      for var, val in bestPoint.items():
+        self.raiseAMessage(finalTemplate.format(name=var, value=val))
+      self.raiseAMessage('*' * 80)
+      # write final best solution to soln export
+      self._updateSolutionExport(bestTraj, self.normalizeData(bestOpt), 'final', 'None')
+    else: #self._isMultiObjective true
+      for i in range(len(self._optPointHistory[traj][-1][0][self._objectiveVar[0]])):
+        opt = self._optPointHistory[traj][-1][0]
```

Comment:
I have a couple of comments here: 
1. Could you justify the differences between the if...else... statement here, except the raiseAMessage? I think they can be combined.
2. Why the message only present in the first if condition, not else condition?
3. In the first 'if' check, the 'opt' variable is not used in other places, and the bestOpt is retrieved in a different way, is there a reason for that?

Reply:
I have simplified the code a bit, but without changing the results, I am not sure that the if can be removed.

Reply:
@joshua-cogliati-inl, if you can merge those and further simplify them, this would be great. I don't think there is a major difference beween how we finalize sampler in single- vs. multi- Objective. If you recall this and the Collect Opt point (and Collect opt point multi) are locations I was wishing to either merge or simplify. Let me know if you think there is a reason, this cannot be further simplified.

Reply:
I did finally manage to merge them (it too some modification of parts that were already shared).

Reply:
https://github.com/idaholab/raven/pull/2326/commits/e6ac5a5010200b26c5a9d8ea3a91161cafb5736a